core/CronArchive.php

```diff
@@ -494,6 +494,15 @@ private function launchArchivingFor($archives, QueueConsumer $queueConsumer)
             $checkInvalid = $this->checkResponse($content, $url);
 
             $stats = json_decode($content, $assoc = true);
+            // If decoding failed and no errors were found, try to remove all the log messages before the JSON response
+            if (!is_array($stats) && $checkInvalid) {
+                $jsonStartPos = strpos($content, '{"idarchives:');
+                if ($jsonStartPos !== false && $jsonStartPos > 0) {
+                    $cleanedContent = substr($content, $jsonStartPos);
+                    // Try decoding again in case we were able to isolate the JSON
+                    $stats = json_decode($cleanedContent, true);
+                }
+            }
```

Comment:
I added this because even when there weren't any errors the `Error unserializing the following response from ...` error was getting thrown because the API output was including all of the log messages on top of the JSON response. So, if no errors were found, this strips off all of the output before the JSON response body so that it can be decoded correctly.

Reply:
To me this looks more like a workaround than a real fix for the problem. In my opinion API responses should never contain any kind of php error / warning / notice output. If this is can be the case, we might need to consider to optimize the error handling. What exactly is your local log configuration? Do you have the screen write enabled?

Reply:
Yeah. I'm not really sure why the `CoreAdminHome.archiveReports` API method returns a bunch of log messages as part of the response instead of just the JSON response sometimes. I wasn't sure if that was by design for some reason. When I tried calling the API endpoint directly, I received a normal JSON response. However, when I had a cron run the archiving process, I received the following:
```
ERROR [2023-03-03 04:00:03] 2732653 api.local-matomo.cloud  Got invalid response from API request: ?module=API&method=CoreAdminHome.archiveReports&idSite=1&period=day&date=2023-03-03&format=json&trigger=archivephp. Response was '[2023-03-03 04:00:02] piwik.DEBUG: Loaded plugins: CoreVue, CorePluginsAdmin, CoreAdminHome, CoreHome, WebsiteMeasurable, IntranetMeasurable, Diagnostics, CoreVisualizations, Proxy, API, Widgetize, Transitions, LanguagesManager, Actions, Dashboard, MultiSites, Referrers, UserLanguage, DevicesDetection, Goals, Ecommerce, SEO, Events, UserCountry, GeoIp2, VisitsSummary, VisitFrequency, VisitTime, VisitorInterest, RssWidget, Feedback, Monolog, Login, TwoFactorAuth, UsersManager, SitesManager, Installation, CoreUpdater, CoreConsole, ScheduledReports, UserCountryMap, Live, PrivacyManager, ImageGraph, Annotations, MobileMessaging, Overlay, SegmentEditor, Insights, Morpheus, Contents, BulkTracking, Resolution, DevicePlugins, Heartbeat, Intl, Marketplace, UserId, Tour, PagePerformance, CustomDimensions, TagManager, AbTesting, ActivityLog, AdvertisingConversionExport, AnonymousPiwikUsageMeasurement, Bandwidth, CDN, Cloud, Cohorts, ConnectAccounts, CustomAlerts, CustomReports, CustomVariables, FormAnalytics, Funnels, GoogleAnalyticsImporter, HeatmapSessionRecording, LoginSaml, MediaAnalytics, MultiChannelConversionAttribution, ProxySite, RollUpReporting, SearchEngineKeywordsPerformance, SEOWebVitals, ServerFileSync, UsersFlow, VisitorGenerator, WhiteLabel, WooCommerceAnalytics [] {"class":"Piwik\\Plugin\\Manager","request_id":"2732715 api.local-matomo.cloud"}

[2023-03-03 04:00:02] piwik.DEBUG: Archiving forced to trigger for [idSite = 1, period = day 2023-03-03,2023-03-03, segment = , plugin = , report = ]. [] {"class":"CoreAdminHome","request_id":"2732715 api.local-matomo.cloud"}

[2023-03-03 04:00:02] piwik.INFO: initiating archiving via core:archive for [idSite = 1, period = day 2023-03-03,2023-03-03, segment = , plugin = , report = ] [] {"class":"CoreAdminHome","request_id":"2732715 api.local-matomo.cloud"}
...
```

Reply:
@snake14 would you mind sharing you log configuration used when the error occurred?

Reply:
@sgiehl Sure. In my global.ini.php file, I had:
```
[log]
; possible values for log: screen, database, file, errorlog, syslog
log_writers[] = screen

; log level, everything logged w/ this level or one of greater severity
; will be logged. everything else will be ignored. possible values are:
; ERROR, WARN, INFO, DEBUG
; this setting will apply to every log writer, if there is no specific log level defined for a writer.
log_level = WARN
```

In my common.config.ini.php I had:
```
[log]
log_writers[] = screen
log_writers[] = file
```

In my crontab, I had the following:
```
MAILTO=jacob@innocraft.com

*/5 * * * * php /home/jacobr/projects/cloud/src/console core:archive --matomo-domain="https://api.local-matomo.cloud" >> /home/jacobr/cloud_archiving.log
```

Reply:
Ok. I guess the screen log writer causes the trouble there. We should maybe in general disable the screen log writer in API requests or convert possible PHP errors into API errors, so the response is at least valid and parsible. Might be good to create a new issue for that, as we might need to discuss the most suitable solution...

Reply:
Thanks @sgiehl . That makes sense. I'll see if I can close this PR and create a core issue for the API.